ddtrace/internal/native/__init__.py

```diff
@@ -0,0 +1,23 @@
+import os
+import sys
+from typing import Dict
+
+from ._native import DDSketch  # noqa: F401
+from ._native import PyConfigurator
+
+
+def get_configuration_from_disk(debug_logs: bool = False, file_override="") -> Dict[str, str]:
+    """
+    Retrieves the tracer configuration from disk. Calls the PyConfigurator object
+    to read the configuration from the disk using the libdatadog shared library
+    and returns the corresponding configuration
+    """
+    configurator = PyConfigurator(debug_logs)
+    configurator.set_envp(["%s=%s" % (k, v) for k, v in os.environ.items()])
+    configurator.set_args(sys.argv)
+
+    # Set the file override if provided. Only used for testing purposes.
+    if file_override:
+        configurator.set_file_override(file_override)
+
+    return configurator.get_configuration()
```

Comment:
Does this configurator handle sending instrumentation telemetry? If so we may end up sending duplicate instrumentation telemetry events ([here for dd_service](https://github.com/DataDog/dd-trace-py/blob/v2.20.0rc1/ddtrace/settings/config.py#L483)). 

Also it would be nice if we could distinguish between configurations set by customer's via env_vars and configurations that are set by stable configs. We could address this limitation in a future PR.

Reply:
It doesn't afaik, at least it wasn't added in the [libdatadog PR](https://github.com/DataDog/libdatadog/pull/807). 

I agree that we need to display the source here; especially as in the very near future we'll have two sources for stable config (manually set by customer // sent through Fleet Automation). This will be done in another PR, hopefully in the next week or so, and we'll return a list of tuples (key, value, source) instead of a dict. 

It won't matter much as long as we don't have a better way to apply stable configs than env vars, but at least the data will be there for future improvements.